source/common/config/utility.h

```diff
@@ -209,13 +211,50 @@ class Utility {
     return *factory;
   }
 
+  template <class Factory> static Factory& getAndCheckFactoryByType(absl::string_view type) {
+    if (type.empty()) {
+      throw EnvoyException("Provided type for static registration lookup was empty.");
```

Comment:
Is there a test for this condition that verifies this message?

Reply:
Adding several tests to beef up coverage and verify error messages.